pandas/core/series.py

```diff
@@ -2855,6 +2858,9 @@ def _try_kind_sort(arr):
 
         result = self._constructor(arr[sortedIdx], index=self.index[sortedIdx])
 
+        if ignore_index:
+            result = result.reset_index(drop=True)
```

Comment:
Is there a reason for this to differ with the frame implementation? Should be the same in both cases

Reply:
oh, sorry! I thought the other one was faster in the first place, and then later found basically no difference, and forgot to align them to keep consistency.

Changed! thanks! @WillAyd

Reply:
ahh!! i recall why they are different, this `new_data` here is `BlockManager` while in series is not. I revert the change.

Reply:
use ```result.index = ibase.default_index(len(indexer))```

this is NOT a BlockManager here, but a series

Reply:
ahh, sorry for my bad English, i meant `new_data` in `frame.py` is BlockManager (should have used `there` other than `here` to be precise 😅 ), here it is `Series`. @jreback

Reply:
you will see my comment in the PR about drop_duplicates. We do not want to use .reset_index directly *at all*. This causes yet another copy of the data which we can easily avoid here.

you have to be careful to avoid mutating things though.